portal-kernel/src/com/liferay/portal/kernel/servlet/ServletInputStreamAdapter.java

```diff
@@ -68,6 +85,13 @@ public void reset() throws IOException {
 		inputStream.reset();
 	}
 
+	/**
+	* @since Servlet 3.1
+	*/
+	@Override
+	public void setReadListener(ReadListener listener) {
+	}
+
```

Comment:
@holatuwol The requirements from the [Javadoc for this method](https://docs.oracle.com/javaee/7/api/javax/servlet/ServletInputStream.html#setReadListener-javax.servlet.ReadListener-) have not been met.